server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Session.scala

```diff
@@ -31,30 +35,58 @@ import org.apache.james.jmap.model.State.State
 case class IsPersonal(value: Boolean)
 case class IsReadOnly(value: Boolean)
 
-object Account {
-  def apply(name: Username,
-            isPersonal: IsPersonal,
-            isReadOnly: IsReadOnly,
-            accountCapabilities: Set[_ <: Capability]): Account = {
+object AccountId {
+  def from(username: Username): Either[IllegalArgumentException, AccountId] = {
+    val sha1String = Hashing.sha1()
+      .hashString(username.asString(), StandardCharsets.UTF_8)
+      .toString
+    val refinedId: Either[String, Id] = refineV(sha1String)
 
-    new Account(name, isPersonal, isReadOnly, accountCapabilities)
+    refinedId match {
+      case Left(errorMessage: String) => Left(new IllegalArgumentException(errorMessage))
+      case Right(id) => Right(AccountId(id))
+    }
   }
 }
 
-final case class Account private(name: Username,
+final case class AccountId(id: Id)
+
+object Account {
+  val NAME = "name";
+  val IS_PERSONAL = "isPersonal"
+  val IS_READ_ONLY = "isReadOnly"
+  val ACCOUNT_CAPABILITIES = "accountCapabilities"
+
+  def from(name: Username,
+           isPersonal: IsPersonal,
+           isReadOnly: IsReadOnly,
+           accountCapabilities: Set[_ <: Capability]): Either[IllegalArgumentException, Account] =
+    AccountId.from(name) match {
+      case Left(ex: IllegalArgumentException) => Left(ex)
+      case Right(accountId) => Right(new Account(accountId, name, isPersonal, isReadOnly, accountCapabilities))
+    }
+
+  def unapplyIgnoreAccountId(account: Account): Some[(Username, IsPersonal, IsReadOnly, Set[_ <: Capability])] =
+    Some(account.name, account.isPersonal, account.isReadOnly, account.accountCapabilities)
+}
+
+final case class Account private(accountId: AccountId,
+                                 name: Username,
                                  isPersonal: IsPersonal,
                                  isReadOnly: IsReadOnly,
                                  accountCapabilities: Set[_ <: Capability])
 
 object State {
+  val INSTANCE: State = "000001"
```

Comment:
private?

Reply:
I need to access it from other place. Even though in the same class, but private seems prevent this to be imported

Reply:
please look at visibility in Scala, you are not restricted to public and private

Reply:
`package` is the only possible solution I found. Can I use it?

Reply:
i would use  the visibility `private[model]` to make the value usable in the package model.
in scala you can add qualifier to the private visibility. 
an enclosing scope like `private[model]` or `private[jmap]`

you can also restrict the visibility more than `private` with `private[this]` which restrict the access to the instance.

Reply:
that's cool! though